application-admintools-default/src/main/java/com/xwiki/admintools/internal/downloads/DownloadsManager.java

```diff
@@ -0,0 +1,246 @@
+    {
+        return usedFilesResourceProvider.getConfigurationFileContent(fileType, currentServer.retrieveXwikiCfgPath());
+    }
+
+    /**
+     * Checks the admin rights of the calling user.
+     *
+     * @return {@link Boolean} true if the user is admin, false otherwise
+     */
+    public boolean isAdmin()
+    {
+        XWikiContext wikiContext = xcontextProvider.get();
+        DocumentReference user = wikiContext.getUserReference();
+        WikiReference wikiReference = wikiContext.getWikiReference();
+        return this.authorizationManager.hasAccess(Right.ADMIN, user, wikiReference);
+    }
+
+    /**
+     * Retrieve the selected files from the request and create an archive containing them.
+     *
+     * @return {@link Byte} array representing the files archive.
+     */
+    public byte[] downloadMultipleFiles()
+    {
+        XWikiContext wikiContext = xcontextProvider.get();
+        XWikiRequest xWikiRequest = wikiContext.getRequest();
+        Map<String, String[]> files = xWikiRequest.getParameterMap();
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+        try (ZipOutputStream zipOutputStream = new ZipOutputStream(byteArrayOutputStream)) {
+            createArchiveEntries(files, zipOutputStream);
+            zipOutputStream.flush();
+            byteArrayOutputStream.flush();
+            zipOutputStream.close();
+            byteArrayOutputStream.close();
+            return byteArrayOutputStream.toByteArray();
+        } catch (Exception e) {
+            logger.warn("Failed to download logs. Root cause is: [{}]", ExceptionUtils.getRootCauseMessage(e));
+            return null;
+        }
+    }
+
+    /**
+     * Select the right server downloader and call the logs retrieval function.
+     *
+     * @return {@link Byte} array representing the last "n" logs.
+     * @throws IOException
+     */
+    public byte[] callLogsRetriever() throws IOException
+    {
+        XWikiContext wikiContext = xcontextProvider.get();
+        XWikiRequest xWikiRequest = wikiContext.getRequest();
+        long noLines = Long.parseLong(xWikiRequest.getParameter("noLines"));
+        return usedFilesResourceProvider.retrieveLastLogs(serverPath, noLines);
+    }
+
+    /**
+     * Verify which files are selected, prepare the zip entry and add it to the zip stream.
+     *
+     * @param files {@link Map} representing the files that are to be processed.
+     * @param zipOutputStream {@link ZipOutputStream} where the zip entries are stored.
+     * @throws IOException
+     */
+    private void createArchiveEntries(Map<String, String[]> files, ZipOutputStream zipOutputStream) throws IOException
+    {
+        ZipEntry zipEntry;
+        byte[] buffer;
+        for (String fileType : files.get("files")) {
+            switch (fileType) {
+                case "cfg_file":
+                    zipEntry = new ZipEntry("xwiki.cfg");
+                    zipOutputStream.putNextEntry(zipEntry);
```

Comment:
You are preparing here a `ZipEntry` for each type of file. But there are some common things and some particular. What it is particular is just the file name and the content. So, you can keep these inside the switch statement, and create the ZipEntry + adding it to the `zipOutputStream` at the end of each iteration